apps/console/src/features/console-settings/components/console-roles/console-roles-edit/console-role-permissions.tsx

```diff
@@ -0,0 +1,288 @@
+    APIResourceCollectionPermissionScopeInterface
+} from "../../../models/console-roles";
+import { SelectedPermissionCategoryInterface, SelectedPermissionsInterface } from "../../../models/permissions-ui";
+import transformResourceCollectionToPermissions from "../../../utils/transform-resource-collection-to-permissions";
+import CreateConsoleRoleWizardPermissionsForm from
+    "../create-console-role-wizard/create-console-role-wizard-permissions-form";
+
+/**
+ * Props interface of {@link ConsoleRolePermissions}
+ */
+export interface ConsoleRolePermissionsProps extends LoadableComponentInterface, IdentifiableComponentInterface {
+    /**
+     * Is read only.
+     */
+    isReadOnly?: boolean;
+    /**
+     * All the details about the role.
+     */
+    role: RolesInterface;
+    /**
+     * Handle role update callback.
+     */
+    onRoleUpdate: (activeTabIndex: number) => void;
+    /**
+     * Tab index
+     */
+    tabIndex: number;
+}
+
+/**
+ * Component to render the tenant & organization permission trees.
+ *
+ * @param props - Props injected to the component.
+ * @returns ConsoleRolePermissions component.
+ */
+const ConsoleRolePermissions: FunctionComponent<ConsoleRolePermissionsProps> = (
+    props: ConsoleRolePermissionsProps
+): ReactElement => {
+    const {
+        isReadOnly,
+        role,
+        onRoleUpdate,
+        tabIndex,
+        [ "data-componentid" ]: componentId
+    } = props;
+
+    const dispatch: Dispatch = useDispatch();
+
+    const { t } = useTranslation();
+
+    const { data: tenantAPIResourceCollections } = useGetAPIResourceCollections(true, "type eq tenant", "apiResources");
+
+    const { data: organizationAPIResourceCollections } = useGetAPIResourceCollections(
+        true,
+        "type eq organization",
+        "apiResources"
+    );
+
+    const [ permissions, setPermissions ] = useState<CreateRolePermissionInterface[]>(undefined);
+
+    /**
+     * Process the collection and return the selected permissions.
+     *
+     * @param collection - API resource collection.
+     * @param selectedFeatures - Selected features.
+     * @returns Selected permissions.
+     */
+    const processCollection = (
+        collection: APIResourceCollectionInterface,
+        selectedFeatures: string[]
+    ): SelectedPermissionCategoryInterface | null => {
+        const readPermissions: string[] = collection.apiResources.read.flatMap(
+            (resource: APIResourceCollectionPermissionCategoryInterface) =>
+                resource.scopes.map((scope: APIResourceCollectionPermissionScopeInterface) => scope.name)
+        );
+
+        const writePermissions: string[] = collection.apiResources.write.flatMap(
+            (resource: APIResourceCollectionPermissionCategoryInterface) =>
+                resource.scopes.map((scope: APIResourceCollectionPermissionScopeInterface) => scope.name)
+        );
+
+        const hasReadPermissions: boolean = readPermissions.every((scope: string) =>
```

Comment:
Shall we use consistent naming for permissions/scopes here?

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/5362/commits/2e1753fb9574b6d6dc7c8674fed685e7e7638ce9